### PR TITLE
Upgrade selected package dependencies

### LINKED ISSUE #40: Upgrade build_config to ^0.3.0
I'm getting this from pub upgrade on a project:

```console
Because build_runner >=0.8.10 depends on build_config ^0.3.0 and sass_builder >=2.0.0 depends on build_config ^0.2.6, build_runner >=0.8.10 is incompatible with sass_builder
  >=2.0.0.
And because angular_components >=0.9.0-alpha+15 depends on sass_builder ^2.0.2, build_runner >=0.8.10 is incompatible with angular_components >=0.9.0-alpha+15.
So, because __projectName__ depends on both angular_components ^0.9.0-alpha+15 and build_runner ^0.8.10, version solving failed.
```

cc @nshahan @kevmoo @kwalrath

### PR DESCRIPTION
**Pull request by chalin**
Fixes #40 

cc @kevmoo @nshahan @kwalrath

### CODE REVIEW

File: pubspec.yaml
Comment (natebosch) [inline pubspec.yaml:17]: [optional] This would not have been breaking for this usage, so it would be safe to use `">=0.2.6 <0.4.0"`

That said, pretty much everyone is going to want the `0.3.x` versions anyway so it isn't likely to make much difference.

**Review by natebosch — APPROVED** [review]

**Comment by chalin** [comment]
@natebosch - anything else to be done before merging this?

**Comment by natebosch** [comment]
I think this is good to go. Need to wait for someone with access to merge it. I'm not sure if @nshahan is working today.

**Comment by luisvt** [comment]
published also

**Comment by natebosch** [comment]
Thanks!

**Pull request merged**
